pennylane/_device.py

```diff
@@ -729,6 +729,10 @@ def batch_transform(self, circuit):
                     "Can only return the expectation of a single Hamiltonian observable"
                 ) from e
 
+        if len(circuit._obs_sharing_wires) > 0 and not hamiltonian_in_obs:
```

Comment:
@Qottmann you can get rid of this warning by disabling it at the top of the file. Below the module docstring, simply add

```python
# pylint: disable=protected-access
```